Services/AzureManagementAdapter/AzureManagementAdapter.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Exceptions;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Http;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.AzureManagementAdapter
+{
+    public interface IAzureManagementAdapterClient
+    {
+        Task<MetricsResponseModel> PostAsync(string token, MetricsRequestsModel requests);
+    }
+
+    public class AzureManagementAdapterClient : IAzureManagementAdapterClient
+    {
+        private const string DATE_FORMAT = "yyyy-MM-ddTHH:mm:ssZ";
+        private const string METRICS_API_VERSION = "2016-06-01";
+        private const bool ALLOW_INSECURE_SSL_SERVER = true;
+        private readonly IHttpClient httpClient;
+        private readonly ILogger log;
+        private readonly IServicesConfig config;
+        private readonly IDeploymentConfig deploymentConfig;
+        private readonly IDiagnosticsLogger diagnosticsLogger;
+
+        public AzureManagementAdapterClient(
+            IHttpClient httpClient,
+            IServicesConfig config,
+            IDeploymentConfig deploymentConfig,
+            ILogger logger,
+            IDiagnosticsLogger diagnosticsLogger)
+        {
+            this.httpClient = httpClient;
+            this.log = logger;
+            this.config = config;
+            this.deploymentConfig = deploymentConfig;
+            this.diagnosticsLogger = diagnosticsLogger;
+        }
+
+        public async Task<MetricsResponseModel> PostAsync(string token, MetricsRequestsModel requests)
+        {
+            if (requests == null)
+            {
+                requests = this.GetMetricsRequests();
+            }
+
+            var request = this.PrepareRequest($"batch?api-version={this.config.AzureManagementAdapterApiVersion}", token, requests);
+            var response = await this.httpClient.PostAsync(request);
```

Comment:
try..catch

Reply:
`SendAysnc` in `HttpClient.cs` will transform `exception` to `new HttpResponse()`.